MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.PasswordBox.xaml

```diff
@@ -406,6 +408,13 @@
     <Setter Property="wpf:HintAssist.Foreground" Value="{DynamicResource PrimaryHueMidBrush}" />
     <Setter Property="wpf:TextFieldAssist.TextBoxViewMargin" Value="{x:Static wpf:Constants.DefaultTextBoxViewMargin}" />
     <Setter Property="wpf:TextFieldAssist.UnderlineBrush" Value="{DynamicResource PrimaryHueMidBrush}" />
+    <Setter Property="wpf:StylizedBehaviors.Behaviors">
+      <Setter.Value>
+        <wpf:StylizedBehaviorCollection>
+          <internalbehaviors:PasswordBoxBehavior />
+        </wpf:StylizedBehaviorCollection>
+      </Setter.Value>
+    </Setter>
```

Comment:
@Keboo This addition here has me a little concerned. Basically we now always apply the behavior on a `PasswordBox` in order to be able to support bindings via the `PasswordBoxAssist.Password` attached property. However, in doing so the consequence is that if you're using MDIX and just want a `PasswordBox` **without** using a binding, then the clear text password is still exposed in the value of that attached property (because the behavior wires it up). So there is really no way to opt-out of it.

Another approach would be not to add it to the "non-reveal" `Style` (NB. it is required in the reveal `Style` for the reveal `TextBox` to sync), and then let the user opt-in for it. Either by setting yet another attached property, or by directly adding the behavior; the latter would then require the behavior to be public.

Reply:
This is a good point. I think I like the idea of making the user opt-in to using it on the non-reveal style. I have a slight preference to going the attached property route only because I have seen users struggle with nested elements in XAML.

Reply:
I went with a slightly different approach which I think/hope works better. Now the default is to **not** wire up the event handler for PasswordChanged **unless** one of two things:

- The user of the control has set the `PasswordBoxAssist.Password` attached property using a binding, or
- The internal suppression of the above guard is applied. This is used for the "reveal" style where the listener must be wired up for the reveal `TextBox` to sync up correctly.

This allows us to keep the original syntax and not require additional properties from an end users perspective.

Reply:
I think this is a very clean solution. Nice work!

Reply:
Thanks